core/tests/protractor_utils/AdminPage.js

```diff
@@ -272,6 +312,151 @@ var AdminPage = function() {
       expect(name).toEqual(expectedUsernamesArray[j]);
     }
   };
+
+  this.getMiscTab = async function() {
+    await miscTabButton.click();
+    await waitFor.pageToFullyLoad();
+  };
+
+  this.uploadTopicSimilarities =
+    async function(relativePathToSimilaritiesFile, isValidFile) {
+      var absPath = path.resolve(__dirname, relativePathToSimilaritiesFile);
+      await waitFor.visibilityOf(chooseSimilarityFileInput,
+        'Similarity upload form is not visible');
+      await chooseSimilarityFileInput.sendKeys(absPath);
+      await waitFor.elementToBeClickable(similarityFileUploadButton,
+        'Upload button taking too long to be clickable');
+      await similarityFileUploadButton.click();
+      if (isValidFile) {
+        await waitFor.visibilityOf(statusMessage,
+          'Status message not visible');
+        await waitFor.textToBePresentInElement(statusMessage,
+          'Topic similarities uploaded successfully.');
+        expect(await statusMessage.getText()).toEqual(
+          'Topic similarities uploaded successfully.');
+      } else {
+        var text = await statusMessage.getText();
+        await waitFor.visibilityOf(statusMessage,
+          'Status message not visible');
+        await waitFor.textToBePresentInElement(statusMessage,
+          text, 'Text not showing up in status message');
+        expect(await statusMessage.getText()).toEqual(text);
```

Comment:
This check seems circular. Since we get `text` from `statusMessage`, what's the point in expecting that they're equal? I don't see how this would ever fail

Reply:
I've set `text` manually, so the test isn't circular anymore